### PR TITLE
Add query helper on URL

### PR DESCRIPTION
**Pull request by steffendsommer**
Added:
- Small helper on `URL` to make it easier to move query items across urls.
- Bumped to Swift 4.2
- Fixed Swiftlint errors

### CODE REVIEW

File: Sources/Sugar/Helpers/URL.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        guard var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+
+        let currentItems = components.queryItems ?? []
+        components.queryItems = currentItems + item
+        return components.url
+    }
+
+    public func addQueryItems(from URL: URL) -> URL? {
```
Comment (siemensikkema) [inline Sources/Sugar/Helpers/URL.swift]:
```suggestion
    public func addQueryItems(from url: URL) -> URL {
```

File: Sources/Sugar/Helpers/URL.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        guard
+            let items = URLComponents(url: URL, resolvingAgainstBaseURL: false)?.queryItems
+        else {
+            return nil
```
Comment (siemensikkema) [inline Sources/Sugar/Helpers/URL.swift]:
```suggestion
            return self
```

File: Sources/Sugar/Helpers/URL.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+public extension URL {
+    public func addQueryItems(_ item: [URLQueryItem]) -> URL? {
```
Comment (siemensikkema) [inline Sources/Sugar/Helpers/URL.swift]:
```suggestion
    public func addQueryItems(_ item: [URLQueryItem]) -> URL {
```

File: Sources/Sugar/Helpers/URL.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        guard var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else {
+            return nil
```
Comment (siemensikkema) [inline Sources/Sugar/Helpers/URL.swift]:
```suggestion
            return self
```

File: Sources/Sugar/Helpers/URL.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        }
+
+        let currentItems = components.queryItems ?? []
+        components.queryItems = currentItems + item
+        return components.url
```
Comment (siemensikkema) [inline Sources/Sugar/Helpers/URL.swift]:
```suggestion
        return components.url ?? self
```

File: Sources/Sugar/Helpers/URL.swift [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    public func addQueryItems(from URL: URL) -> URL? {
+        guard
+            let items = URLComponents(url: URL, resolvingAgainstBaseURL: false)?.queryItems
```
Comment (siemensikkema) [inline Sources/Sugar/Helpers/URL.swift]:
```suggestion
            let items = URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems
```

**Review by siemensikkema — COMMENTED** [review]
I made some suggestions that make the URL query stuff non-optional. See if you like it :)

**Comment by steffendsommer** [comment]
Thanks for the suggestions @siemensikkema 👍 I do like that it doesn't throw or return an optional, however it won't be nice if someone calls this and it silently fails. Not sure how likely that is to happen, so I think we should get this in and then we can always re-evaluate if it becomes an issue. Feel free to re-review/approve.

**Review by siemensikkema — APPROVED** [review]

**Pull request merged**